### PR TITLE
Add initial singularity support

### PR DESCRIPTION
**Pull request by bgruening**
@jmchilton this add singularity to SLRUM, we can add many more later.

I have tried to unify the macros even more, but ended up with the current solution, it was getting to ugly with nested variables and so on. However, I ended up with `container_type` and `force_container` to make it more general.

### CODE REVIEW

**Comment by bgruening** [comment]
Please note that it is targeting your branch.

File: defaults/main.yml [hunk 1]
```diff
@@ -79,7 +79,8 @@ galaxy_singularity_sudo: false
 # ToDo create default image
 galaxy_singularity_default_image: ''
 galaxy_singularity_volumes_from: ""
-galaxy_singularity_volumes : "$defaults"
+# rw directories are not considered if the parent is ro
+galaxy_singularity_volumes : "$tool_directory:ro,$working_directory:rw,$default_file_path:rw"
```
Comment (jmchilton) [inline defaults/main.yml]: xref https://github.com/galaxyproject/galaxy/pull/4180 - does that fix the need for this?

**Comment by bgruening** [comment]
@jmchilton ok to merge?

**Comment by jmchilton** [comment]
Truly awesome - thanks so much @bgruening!

**Pull request merged**